### PR TITLE
Fix issue with build time config in dev mode

### LINKED ISSUE #9296: elytron-security-jdbc does not use attribute-mappings
**Describe the bug**
Role is not mapped from database to the Elytron realm

**Expected behavior**
Role should be mapped from the database to the security principal

**Actual behavior**
The configuration in `security-jdbc-quickstart` does not extract the `role` column from the database.
When debugging `io.quarkus.elytron.security.jdbc.PrincipalQueryConfig#attributeMappings` never gets populated.

**To Reproduce**
1. Download https://github.com/quarkusio/quarkus-quickstarts/tree/master/security-jdbc-quickstart
2. Follow the instructions to start the database and application
3. `curl -i -X GET -u user:user http://localhost:8080/api/users/me`

**Configuration**
Same as [here](https://github.com/quarkusio/quarkus-quickstarts/blob/master/security-jdbc-quickstart/src/main/resources/application.properties)

**Environment (please complete the following information):**
 - Linux intrepid 5.4.0-29-generic #33-Ubuntu SMP Wed Apr 29 14:32:27 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux
 - openjdk version "11.0.7" 2020-04-14
OpenJDK Runtime Environment (build 11.0.7+10-post-Ubuntu-3ubuntu1)
OpenJDK 64-Bit Server VM (build 11.0.7+10-post-Ubuntu-3ubuntu1, mixed mode, sharing)
 - Quarkus 1.4.2.Final
 - Gradle 6.4

### LINKED ISSUE #9415: Flyway with "multiple datasources" does not migrate at start
**Describe the bug**
There are two datasources that should be be migrated on startup by Flyway, but the scripts are not executed. From the logs it can bee seen that the directories are recognized correctly by the Flyway extension, but not applied later. This has been working before (~1.3.x) and stopped working with current master source.

Example (database/mycompany)
```
Adding application migrations in path 'quarkus-quickstarts/hibernate-orm-quickstart-multi-tenancy/target/classes/database/mycompany/' using protocol 'file'
Database: jdbc:postgresql://localhost:5433/mycompany
Current version of schema "mycompany": null
Schema "mycompany" is up to date. No migration necessary.
```

**Expected behavior**
The scripts in "src/main/resources/database" should be used for migration on startup.

**Actual behavior**
The scripts in "src/main/resources/database" are not applied on startup.

**To Reproduce**
1. Checkout Quickstart fork: https://github.com/michael-schnell/quarkus-quickstarts/
2. Change into [hibernate-orm-quickstart-multi-tenancy](https://github.com/michael-schnell/quarkus-quickstarts/tree/master/hibernate-orm-quickstart-multi-tenancy)
3. Execute "DATABASE Multitenancy" example as described in [README.md](https://github.com/michael-schnell/quarkus-quickstarts/blob/master/hibernate-orm-quickstart-multi-tenancy/README.md#database-multitenancy)
4. Check the log output: The two datasource schemas "base" and "mycompany" are not applied

**Configuration**
See https://github.com/michael-schnell/quarkus-quickstarts/blob/master/hibernate-orm-quickstart-multi-tenancy/src/main/resources/application.properties

**Log**
```
2020-05-18 16:04:33,275 INFO  [io.qua.fly.FlywayProcessor] (build-61) Adding application migrations in path 'quarkus-quickstarts/hibernate-orm-quickstart-multi-tenancy/target/classes/db/migration/' using protocol 'file'
2020-05-18 16:04:33,277 INFO  [io.qua.fly.FlywayProcessor] (build-61) Adding application migrations in path 'quarkus-quickstarts/hibernate-orm-quickstart-multi-tenancy/target/classes/database/mycompany/' using protocol 'file'
2020-05-18 16:04:33,291 INFO  [io.qua.fly.FlywayProcessor] (build-61) Adding application migrations in path 'quarkus-quickstarts/hibernate-orm-quickstart-multi-tenancy/target/classes/database/base/' using protocol 'file'
2020-05-18 16:04:33,291 INFO  [io.qua.fly.FlywayProcessor] (build-61) Adding application migrations in path 'quarkus-quickstarts/hibernate-orm-quickstart-multi-tenancy/target/classes/db/migration/' using protocol 'file'
2020-05-18 16:04:36,831 INFO  [io.agr.pool] (Quarkus Main Thread) Datasource '<default>': Initial size smaller than min. Connections will be created when necessary
2020-05-18 16:04:36,835 INFO  [io.agr.pool] (Quarkus Main Thread) Datasource 'mycompany': Initial size smaller than min. Connections will be created when necessary
2020-05-18 16:04:36,840 INFO  [io.agr.pool] (Quarkus Main Thread) Datasource 'base': Initial size smaller than min. Connections will be created when necessary
2020-05-18 16:04:36,938 INFO  [org.fly.cor.int.lic.VersionPrinter] (Quarkus Main Thread) Flyway Community Edition 6.4.2 by Redgate
2020-05-18 16:04:37,092 INFO  [org.fly.cor.int.dat.DatabaseFactory] (Quarkus Main Thread) Database: jdbc:postgresql://localhost:5433/mycompany (PostgreSQL 12.2)
2020-05-18 16:04:37,123 INFO  [org.fly.cor.int.com.DbMigrate] (Quarkus Main Thread) Current version of schema "mycompany": null
2020-05-18 16:04:37,124 INFO  [org.fly.cor.int.com.DbMigrate] (Quarkus Main Thread) Schema "mycompany" is up to date. No migration necessary.
2020-05-18 16:04:37,140 INFO  [org.fly.cor.int.dat.DatabaseFactory] (Quarkus Main Thread) Database: jdbc:postgresql://localhost:5432/quarkus_test (PostgreSQL 12.2)
2020-05-18 16:04:37,153 INFO  [org.fly.cor.int.com.DbMigrate] (Quarkus Main Thread) Current version of schema "public": << Empty Schema >>
2020-05-18 16:04:37,153 INFO  [org.fly.cor.int.com.DbMigrate] (Quarkus Main Thread) Schema "public" is up to date. No migration necessary.
2020-05-18 16:04:37,685 INFO  [io.quarkus] (Quarkus Main Thread) hibernate-orm-quickstart-multi-tenancy 1.0-SNAPSHOT on JVM (powered by Quarkus 999-SNAPSHOT) started in 6.337s. Listening on: http://0.0.0.0:8080
2020-05-18 16:04:37,687 INFO  [io.quarkus] (Quarkus Main Thread) Profile database activated. Live Coding activated.
2020-05-18 16:04:37,688 INFO  [io.quarkus] (Quarkus Main Thread) Installed features: [agroal, cdi, flyway, hibernate-orm, jdbc-postgresql, mutiny, narayana-jta, resteasy, resteasy-jsonb]
^C2020-05-18 16:04:42,798 INFO  [io.quarkus] (Quarkus Main Thread) hibernate-orm-quickstart-multi-tenancy stopped in 0.029s

```
**Environment (please complete the following information):**
 - Linux quarkus-vm 4.15.0-20-generic #21-Ubuntu SMP Tue Apr 24 06:16:15 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
 - openjdk version "11.0.6" 2020-01-14
 - OpenJDK Runtime Environment GraalVM CE 20.0.0 (build 11.0.6+9-jvmci-20.0-b02)
 - OpenJDK 64-Bit Server VM GraalVM CE 20.0.0 (build 11.0.6+9-jvmci-20.0-b02, mixed mode, sharing)
 - Quarkus: Commit [87522320380e9f201a2f4c027f3e50bc1ffd6f06](https://github.com/quarkusio/quarkus/commit/87522320380e9f201a2f4c027f3e50bc1ffd6f06)
 - apache-maven-3.6.2

**Additional context**
I'm currently preparing a quickstart for the new Hibernate multi-tenancy feature. This example was already working fine with 999-SNAPSHOT from source ~1.3.x, but it fails now with current Quarkus master source. The problem is actually not related to the new feature as it happens before anything from that code is executed. The problem is just the missing Flyway migration. It seems to be something that changed between ~1.3.x and the current master most likely in the Flyway extension.

### PR DESCRIPTION
**Pull request by stuartwdouglas**
Fixes #9296
Fixes #9415

### CODE REVIEW

File: core/deployment/src/main/java/io/quarkus/deployment/configuration/RunTimeConfigurationGenerator.java [hunk 1]
```diff
@@ -783,6 +736,34 @@ public void run() {
             generateDefaultValuesConfigSourceClass(buildTimeRunTimePatternMap, BTRTDVCS_CLASS_NAME);
         }
 
+        private static void configSweepLoop(MethodDescriptor siParserBody, MethodCreator clinit2, ResultHandle clinitConfig2) {
```
Comment (geoand) [inline core/deployment/src/main/java/io/quarkus/deployment/configuration/RunTimeConfigurationGenerator.java:739]: Minor nitpick: The arguments names all need to be made more general I think :)

**Comment by geoand** [comment]
Tested this PR with #9415 and it works like a charm!

Fixes seems reasonable to me, :+1:

File: core/deployment/src/main/java/io/quarkus/deployment/configuration/RunTimeConfigurationGenerator.java (hunk 1 above)
Comment (stuartwdouglas) [reply core/deployment/src/main/java/io/quarkus/deployment/configuration/RunTimeConfigurationGenerator.java]: Oops, this was an intellij automatic refactoring and I did not look at the method names.

**Review by geoand — APPROVED** [review]
:+1: 

I looked at the generated bytecode for both prod mode and dev-mode and I understand the change.

Thanks a lot for looking into it!

**Comment by geoand** [comment]
We'll need to restart CI, the failures seems very weird and completely unrelated to the PR (I tried the Spring tests locally from this PR and they worked just fine).

**Comment by geoand** [comment]
I pushed a commit with the test I added for Flyway (since CI needed a restart anyway)

**Comment by geoand** [comment]
quay.io seems to still have problems :(

**Pull request merged**